### PR TITLE
Add windows support for python

### PR DESCRIPTION
**Pull request by qiaozha**
This PR is meant to fix the virtual env folder structure in windows problem.
In windows system, the executable binary files is venv/Scripts/a.exe

### CODE REVIEW

**Comment by qiaozha** [comment]
@elijahr Could you take a look at this PR and if it looks good to you, can you release a new lintball npm package ?

File: lib/cmds.bash [hunk 1]
```diff
@@ -17,12 +23,18 @@ cmd_autopep8() {
   local mode path
   mode="${1//mode=/}"
   path="${2//path=/}"
+  autopep8=""
+  if [ -f "${LINTBALL_DIR}/tools/python-env/bin/autopep8" ]; then
+    autopep8="${LINTBALL_DIR}/tools/python-env/bin/autopep8";
+  elif [ -f "${LINTBALL_DIR}/tools/python-env/Scripts/autopep8.exe" ]; then
+    autopep8="${LINTBALL_DIR}/tools/python-env/Scripts/autopep8.exe"
+  fi
```
Comment (elijahr) [inline lib/cmds.bash]:
```suggestion
    autopep8="${LINTBALL_DIR}/tools/python-env/Scripts/autopep8.exe"
  else
    echo "Could not find autopep8 executable" >&2
    return 1
  fi 
```

File: lib/cmds.bash [hunk 2]
```diff
@@ -32,12 +44,18 @@ cmd_black() {
   local mode path
   mode="${1//mode=/}"
   path="${2//path=/}"
+  blackexe=""
+  if [ -f "${LINTBALL_DIR}/tools/python-env/bin/black" ]; then
+    blackexe="${LINTBALL_DIR}/tools/python-env/bin/black";
+  elif [ -f "${LINTBALL_DIR}/tools/python-env/Scripts/black.exe" ]; then
+    blackexe="${LINTBALL_DIR}/tools/python-env/Scripts/black.exe"
+  fi
```
Comment (elijahr) [inline lib/cmds.bash]:
```suggestion
    blackexe="${LINTBALL_DIR}/tools/python-env/Scripts/black.exe"
  else
    echo "Could not find black executable" >&2
    return 1
  fi  
```

File: lib/cmds.bash [hunk 3]
```diff
@@ -62,12 +80,18 @@ cmd_docformatter() {
   local mode path
   mode="${1//mode=/}"
   path="${2//path=/}"
+  docformatter=""
+  if [ -f "${LINTBALL_DIR}/tools/python-env/bin/docformatter" ]; then
+    docformatter="${LINTBALL_DIR}/tools/python-env/bin/docformatter";
+  elif [ -f "${LINTBALL_DIR}/tools/python-env/Scripts/docformatter.exe" ]; then
+    docformatter="${LINTBALL_DIR}/tools/python-env/Scripts/docformatter.exe"
+  fi
```
Comment (elijahr) [inline lib/cmds.bash]:
```suggestion
    docformatter="${LINTBALL_DIR}/tools/python-env/Scripts/docformatter.exe"
  else
    echo "Could not find docformatter executable" >&2
    return 1
  fi  
```

File: lib/cmds.bash [hunk 4]
```diff
@@ -77,12 +101,18 @@ cmd_isort() {
   local mode path
   mode="${1//mode=/}"
   path="${2//path=/}"
+  isort=""
+  if [ -f "${LINTBALL_DIR}/tools/python-env/bin/isort" ]; then
+    isort="${LINTBALL_DIR}/tools/python-env/bin/isort";
+  elif [ -f "${LINTBALL_DIR}/tools/python-env/Scripts/isort.exe" ]; then
+    isort="${LINTBALL_DIR}/tools/python-env/Scripts/isort.exe"
+  fi
```
Comment (elijahr) [inline lib/cmds.bash]:
```suggestion
    isort="${LINTBALL_DIR}/tools/python-env/Scripts/isort.exe"
  else
    echo "Could not find isort executable" >&2
    return 1
  fi  
```

File: lib/cmds.bash [hunk 5]
```diff
@@ -2,12 +2,18 @@ cmd_autoflake() {
   local mode path
   mode="${1//mode=/}"
   path="${2//path=/}"
+  autoflake=""
+  if [ -f "${LINTBALL_DIR}/tools/python-env/bin/autoflake" ]; then
+    autoflake="${LINTBALL_DIR}/tools/python-env/bin/autoflake"
+  elif [ -f "${LINTBALL_DIR}/tools/python-env/Scripts/autoflake.exe" ]; then
+    autoflake="${LINTBALL_DIR}/tools/python-env/Scripts/autoflake.exe"
+  fi
```
Comment (elijahr) [inline lib/cmds.bash]:
```suggestion
    autoflake="${LINTBALL_DIR}/tools/python-env/Scripts/autoflake.exe"
  else
    echo "Could not find autoflake executable" >&2
    return 1
  fi
```

File: lib/install.bash [hunk 6]
```diff
@@ -61,14 +61,26 @@ install_pip_requirements() {
...
       fi
     fi
-    if [ ! -f "python-env/bin/pip" ]; then
+    local activateexe
+    activateexe=""
+    if [ -f "python-env/bin/activate" ]; then
+      activateexe="python-env/bin/activate"
+    elif [ -f "python-env/Scripts/activate" ]; then
+      activateexe="python-env/Scripts/activate"
+    fi
```
Comment (elijahr) [inline lib/install.bash]:
```suggestion
      activateexe="python-env/Scripts/activate"
    else
      echo "Could not find venv activate script" >&2
      return 1
    fi
```

**Review by elijahr — CHANGES_REQUESTED** [review]
Hello @qiaozha! Thank you so much for this contribution. The change set looks good to me - a few minor suggestions.

**Comment by qiaozha** [comment]
I find there's lint issue in this PR

**Comment by qiaozha** [comment]
I had this commit id to fix it https://github.com/qiaozha/lintball/commit/7e38f0348eae02954715289e444d86072e483681 but the PR already got merged

**Comment by elijahr** [comment]
@qiaozha Yeah, it was just some trailing whitespace in one of the files, I removed it in 3969551c9b9fad5bd59fb01da1f33a56d303582c.

The macOS tests were already failing on the devel branch, not related to your changes. I've disabled them for now and am just running tests with Ubuntu. I will revisit macOS later, I think the bash version is just out of date. 

I'm waiting on tests to pass here: https://github.com/elijahr/lintball/actions/runs/536189558

Then I will tag `v1.2.0` and publish to npm. I'll let you know here once that is done.

**Comment by elijahr** [comment]
@qiaozha I see there was some checks for yamllint - thanks, I'll cherry-pick that commit.

**Comment by qiaozha** [comment]
@elijahr Thank you so much for the quick response

**Comment by elijahr** [comment]
@qiaozha you're welcome. Thanks again for the contribution! I've released 1.2.1 to npm.

**Comment by qiaozha** [comment]
Thanks :)

**Pull request merged**